sql_assign2.sql

```diff
@@ -0,0 +1,183 @@
+
+
+--  List all employees not named 'John' (first name).
+SELECT * FROM Employees WHERE FirstName NOT LIKE 'John'
+
+-- Identify Marketing department employees earning $60,000 or less who were hired after June 30, 2019.
+SELECT * FROM Employees WHERE Department='Marketing' AND Salary <= 60000 AND HireDate > '2021-06-30'
+
+-- Find employees whose first name contains the letters 'an' anywhere and ends with 'e'.
+SELECT * FROM Employees WHERE FirstName LIKE '%an%e'
+
+
+
+
+-- For this assignment, use the following tables:
+-- CREATE TABLE Sales (
+--     SaleID INT PRIMARY KEY,
+--     ProductID INT,
+--     CustomerID INT,
+--     SaleDate DATE,
+--     Quantity INT,
+--     UnitPrice DECIMAL(10, 2)
+-- );
+
+-- CREATE TABLE Products (
+--     ProductID INT PRIMARY KEY,
+--     ProductName VARCHAR(100),
+--     Category VARCHAR(50)
+-- );
+
+-- CREATE TABLE Customers (
+--     CustomerID INT PRIMARY KEY,
+--     CustomerName VARCHAR(100),
+--     City VARCHAR(50),
+--     State VARCHAR(2)
+-- );
+
+
+INSERT INTO Products (ProductID, ProductName, Category) VALUES
+(1, 'Laptop', 'Electronics'),
+(2, 'Smartphone', 'Electronics'),
+(3, 'Tablet', 'Electronics'),
+(4, 'Coffee Maker', 'Home Appliances'),
+(5, 'Desk Chair', 'Furniture'),
+(6, 'Gaming Mouse', 'Accessories'),
+(7, 'Office Desk', 'Furniture'),
+(8, 'Wireless Headphones', 'Electronics'),
+(9, 'Printer', 'Electronics'),
+(10, 'Camera', 'Photography'),
+(11, 'Smartwatch', 'Electronics'),
+(12, 'Air Fryer', 'Home Appliances');
+
+
+
+INSERT INTO Customers (CustomerID, CustomerName, City,State) VALUES
+(1, 'Alice Johnson', 'New York','KL'), 
+(2, 'Bob Smith', 'Los Angeles','HR'),
+(3, 'Charlie Brown', 'Chicago','JK'),
+(4, 'Diana Prince', 'Miami','ER'),
+(5, 'Ethan Hunt', 'San Francisco','PY'),
+(6, 'Fiona Green', 'Houston','XC'),
+(7, 'George Martin', 'Seattle','QW'),
+(8, 'Hannah Baker', 'Boston','NM'),
+(9, 'Ian Wright', 'Phoenix','LO'),
+(10, 'Julia Roberts', 'Dallas','UC'),
+(11, 'Kevin Spacey', 'Austin','ZF'),
+(12, 'Laura Croft', 'Philadelphia','AS');
+
+
+INSERT INTO Sales (SaleID, ProductID, CustomerID, SaleDate, Quantity, UnitPrice) VALUES
+(1, 1, 1, '2023-01-15', 1, 1200.00),
+(2, 2, 2, '2023-01-20', 2, 800.00),
+(3, 3, 3, '2023-01-25', 2, 300.00),
+(4, 4, 4, '2023-01-30', 1, 150.00),
+(5, 5, 5, '2023-02-05', 3, 250.00),
+(6, 6, 6, '2023-02-10', 1, 50.00),
+(7, 7, 7, '2023-02-15', 2, 400.00),
+(8, 8, 8, '2023-02-20', 2, 200.00),
+(9, 9, 9, '2023-02-25', 1, 150.00),
+(10, 10, 10, '2023-03-01', 1, 600.00),
+(11, 11, 11, '2023-03-05', 2, 300.00),
+(12, 12, 12, '2023-03-10', 1, 120.00);
+
+
+-- Calculate the total sales amount for each product. Display the ProductID and total sales amount.
+SELECT ProductID, SUM(Quantity * UnitPrice) AS Total_Sales FROM Sales GROUP BY ProductID
+
+-- Find the average quantity sold per sale for each product category.
+SELECT Category,AVG(Quantity) AS Average_sale FROM Products,Sales WHERE Products.ProductID = Sales.ProductID GROUP BY Category
```

Comment:
which join is this ?

Reply:
What if I have products that are not sold, What will be the output of this query?

Reply:
> which join is this ?

CROSS JOIN with where condition

Reply:
> What if I have products that are not sold, What will be the output of this query?
We can use Products LEFT JOIN Sales ON Products.ProductID = Sales.ProductID